WonderMake/Utilities/Container/Container.h

```diff
@@ -0,0 +1,159 @@
+#pragma once
+#include <concepts>
+#include <type_traits>
+
+#include <vector>
+#include <list>
+
+#include "Utilities/plf_colony.h"
+
+#include "Traits/ContainerTraits.h"
+#include "BackendImplementations/ColonyBackend.h"
+#include "BackendImplementations/LinkedListBackend.h"
+#include "BackendImplementations/VectorBackend.h"
+
+#include "Utilities/TypeTraits/ParameterPack.h"
+#include "Utilities/Container/Traits/Implications.h"
+#include "Utilities/TypeTraits/ContainsType.h"
+#include <iostream>
+
+template<typename TObjectType, typename TContainerBackends, ContainerTrait ... TContainerTraits>
+class ContainerBase
+{
+	//[Nicos]: These need to be at the top so we can use Storage as a type
+private:
+	using TraitsPack = typename ResolvedImplications<TContainerTraits...>::type;
+
+	template <typename ... TBackends>
+	constexpr static auto StorageType(ParameterPack<TBackends...>)
+	{
+		return FirstSatisfyingBackend<TraitsPack, TBackends...>::type();
+	}
+
+	using Storage = decltype(StorageType(TContainerBackends()));
+
+	template <typename T>
+	constexpr static bool UsesStorage = std::is_same_v<Storage, T>;
+
+	Storage myBackend;
+
+public:
+
+	void PrintTraits()
+	{
+		std::cout << typeid(TraitsPack).name() << std::endl;
+	}
```

Comment:
We shouldn't print in base classes, it's better if we return a string or string_view

Reply:
Ya it was temp